mlshvch/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+file = File.open('data.txt', 'r')
+data = file.read
+file.close
+
+EXIT_COMMAND = 'exit!'
+ERROR_MESSAGE = 'City name can not be blank!'
+DEFAULT_VALUE = -1.0
+
+def extract_data(data_string)
+  data_string.split("\n").each_with_object({}) do |data, city_rainfall_stats|
+    town, rainfall_stats = data.split(':')
+    city_rainfall_stats[town] = rainfall_stats.split(',').each_with_object([]) do |results, month_rainfall_stats|
+      month_rainfall_stats << results.split(' ')[1]
```

Comment:
If you need to iterate through collection, performing transformation over the values, and get another collection as a result, it is much easier to use [Enumerable#map](https://devdocs.io/ruby~3/enumerable#method-i-map), like I mentioned before.

```ruby
city_rainfall_stats[town] = rainfall_stats.split(',').map { |month_stat| month_stat.split(' ').last.to_f }
```